File: Python/lab1/WarrantyService.py
class WarrantyService(object):
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(WarrantyService, cls).__new__(cls)

            cls._instance.warranty_info = {}

        return cls._instance

    def register_warranty(self, product_id, warranty_period):
        self.warranty_info[product_id] = warranty_period

    def check_warranty(self, product_id):
        if product_id in self.warranty_info:
            return f"Гарантія на товар {product_id} дійсна до {self.warranty_info[product_id]}"
        else:
            return f"Гарантія на товар {product_id} не знайдена"

File: Java/lab1/src/main/java/org/example/WarrantyService.java
package org.example;
import java.util.HashMap;

public class WarrantyService {
    private static WarrantyService instance;
    private HashMap<String, String> warrantyInfo;
    private WarrantyService() {
        warrantyInfo = new HashMap<>();
    }

    public static WarrantyService getInstance() {
        if (instance == null)
            instance = new WarrantyService();
        return instance;
    }

    public void registerWarranty(String productId, String warrantyPeriod) {
        warrantyInfo.put(productId, warrantyPeriod);
    }

    public String checkWarranty(String productId) {
        if (warrantyInfo.containsKey(productId)) {
            return "Гарантія на товар " + productId + " дійсна до " + warrantyInfo.get(productId);
        } else {
            return "Гарантія на товар " + productId + " не знайдена";
        }
    }
}
